src/pages/_app.js

```diff
@@ -25,6 +26,14 @@ function MyApp({ Component, pageProps }) {
           <ReactQueryDevtools initialIsOpen={false} />
         </Hydrate>
       </QueryClientProvider>
+
+      <NextNProgress
+        color="#0dcaf0"
+        startPosition={0.3}
+        stopDelayMs={200}
+        height={3}
+        showOnShallow={true}
```

Comment:
```suggestion
        showOnShallow
```
Não é necessário o React já entende como true.